src/agents/common/shared_encoder.py

```diff
@@ -0,0 +1,23 @@
+"""
+Shared encoder
```

Comment:
Should we specify as the shared encoder, if its in common isn't it automatically shared?

Reply:
Made more sense to me at the time of implementation.
Though it is in the common folder so you're right